### PR TITLE
docs: migrate to newer :::warning syntax

### PR DESCRIPTION
**Pull request by mxschmitt**
- Migrate from `:::warning` to `:::danger[Warning]`.
- Migrate from `:::caution` to `:::warning`.
- `:::warning` and `:::caution` were the color same before.
- `:::caution` is deprecated and gets removed in v4 of Docusaurus
- Also migrate to the [newer title syntax](https://github.com/facebook/docusaurus/blob/3877c1254f29a64b9f2301775d9ecc2ae5e2a740/packages/docusaurus-utils/src/markdownUtils.ts#L86-L91)

### CODE REVIEW

**Review by dgozman — COMMENTED** [review]
Does this mean all our existing `:::warning` are rendered yellow? Shall we migrate them to `:::danger[Warning]` or something?

File: docs/src/api/class-page.md [hunk 1]
```diff
@@ -3146,7 +3146,7 @@ return value resolves to `[]`.
 ## async method: Page.addLocatorHandler
 * since: v1.42
 
-:::warning Experimental
+:::danger[Warning] Experimental
```
Comment (dgozman) [inline docs/src/api/class-page.md:3149]:
```suggestion
:::warning[Experimental]
```

File: docs/src/api/class-page.md [hunk 2]
```diff
@@ -3160,7 +3160,7 @@ Things to keep in mind:
 * The execution time of the handler counts towards the timeout of the action/assertion that executed the handler. If your handler takes too long, it might cause timeouts.
 * You can register multiple handlers. However, only a single handler will be running at a time. Make sure the actions within a handler don't depend on another handler.
 
-:::warning
+:::danger[Warning]
```
Comment (dgozman) [inline docs/src/api/class-page.md:3163]: This one is fresh in my mind, and it was intended to be yellow. So, perhaps, not all of the warning are actually danger. Maybe you were right to keep them as warnings instead?

```suggestion
:::warning[Warning]
```

**Review by dgozman — APPROVED** [review]

File: docs/src/api/class-page.md [hunk 3]
```diff
@@ -3160,7 +3160,7 @@ Things to keep in mind:
 * The execution time of the handler counts towards the timeout of the action/assertion that executed the handler. If your handler takes too long, it might cause timeouts.
 * You can register multiple handlers. However, only a single handler will be running at a time. Make sure the actions within a handler don't depend on another handler.
 
-:::warning
+:::warning[Warning]
```
Comment (mxschmitt) [inline docs/src/api/class-page.md:3163]:
```suggestion
:::warning
```

File: docs/src/test-parallel-js.md [hunk 4]
```diff
@@ -216,7 +216,7 @@ When you **disable parallel test execution**, Playwright Test runs test files in
 
 ### Use a "test list" file
 
-:::warning
+:::danger[Warning]
```
Comment (mxschmitt) [inline docs/src/test-parallel-js.md:219]:
```suggestion
:::warning
```

**Pull request merged**